cranelift-codegen/build.rs

```diff
@@ -30,29 +25,36 @@ fn main() {
 
     let out_dir = env::var("OUT_DIR").expect("The OUT_DIR environment variable must be set");
     let target_triple = env::var("TARGET").expect("The TARGET environment variable must be set");
-    let cranelift_targets = env::var("CRANELIFT_TARGETS").ok();
-    let cranelift_targets = cranelift_targets.as_ref().map(|s| s.as_ref());
-    let python = identify_python();
 
     // Configure isa targets cfg.
-    match isa_targets(cranelift_targets, &target_triple) {
-        Ok(isa_targets) => {
-            for isa in &isa_targets {
-                println!("cargo:rustc-cfg=build_{}", isa.to_string());
-            }
-        }
-        Err(err) => {
-            eprintln!("Error: {}", err);
-            process::exit(1);
+    let isa_targets = meta::isa::Isa::all()
+        .into_iter()
+        .map(|isa| isa.to_string())
+        .filter(|isa| {
+            let env_key = format!("CARGO_FEATURE_{}", isa.to_uppercase());
```

Comment:
Would it work to move the `to_string()` here into the `filter` body, rather than using a `map` first?

Reply:
Yes, it's nice: the `isa_targets` just is the vector of ISA targets we want in the end, so that removes an entire function `isa_from_targets` in the `meta` crate.